kakao/src/main/kotlin/com/agoda/kakao/screen/Screen.kt

```diff
@@ -27,6 +28,11 @@ open class Screen<out T: Screen<T>>: ScreenActions {
     operator fun invoke(function: T.() -> Unit) = function.invoke(this as T)
 
     companion object {
+        /**
+         * The visibility of rootView will be checked when entering the screen
+         */
+        var rootView: KBaseView<*>? = null
```

Comment:
This property should not be in companion. As I understand your intention, different screens can have different root views. In that case it should be the property of the class:
`open var root: KBaseView<*>? = null`